ansible_runner/interface.py

```diff
@@ -52,10 +52,10 @@ def init_runner(**kwargs):
     '''
     # If running via the transmit-worker-process method, we must only extract things as read-only
     # inside of one of these commands. That could be either transmit or worker.
-    if kwargs.get('streamer') not in ('worker', 'process'):
+    if kwargs.get('process_isolation') or kwargs.get('streamer') not in ('worker', 'process'):
```

Comment:
These lines are what I was really looking for in here. Trying to use direct role execution from the CLI using an execution environment didn't work because of the path assignment behavior.

Reply:
... or I misread the logic.

Reply:
In practice, artifacts shouldn't be getting dumped because of `suppress_env_files`